drgn/commands/_builtin/crash/_runq.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (c) 2025, Oracle and/or its affiliates.
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+"""
+crash runq - Display the tasks on the run queues of each cpu.
+
+Implements the crash "runq" command for drgn
+"""
+import argparse
+from typing import Any, Dict, Iterator, List, Tuple
+
+from drgn import Object, Program
+from drgn.commands import argument, drgn_argument, mutually_exclusive_group
+from drgn.commands.crash import crash_command, parse_cpuspec
+from drgn.helpers.common.format import CellFormat, escape_ascii_string, print_table
+from drgn.helpers.linux.cpumask import for_each_online_cpu
+from drgn.helpers.linux.percpu import per_cpu
+from drgn.helpers.linux.runqueue import rq_for_each_fair_task, rq_for_each_rt_task
+from drgn.helpers.linux.sched import task_rq, task_since_last_arrival_ns
+
+
+def get_rq_per_cpu(prog: Program, cpus: List[int] = []) -> Iterator[Tuple[int, Object]]:
+    """
+    Get runqueue for selected cpus
+
+    :param prog: drgn program
+    :param cpus: a list of int
+    :return: Iterator of (int, ``struct rq``) tuples
+    """
+    online_cpus = list(for_each_online_cpu(prog))
+
+    if cpus:
+        selected_cpus = [cpu for cpu in online_cpus if cpu in cpus]
+    else:
+        selected_cpus = online_cpus
+
+    for cpu in selected_cpus:
+        runqueue = per_cpu(prog["runqueues"], cpu)
+        yield (cpu, runqueue)
+
+
+def timestamp_str(ns: int) -> str:
+    """Convert nanoseconds to 'days HH:MM:SS.mmm' string."""
+    ms_total = ns // 1000000
+    secs_total, ms = divmod(ms_total, 1000)
+    mins_total, secs = divmod(secs_total, 60)
+    hours_total, mins = divmod(mins_total, 60)
+    days, hours = divmod(hours_total, 24)
+
+    return f"{days} {hours:02}:{mins:02}:{secs:02}.{ms:03}"
+
+
+@crash_command(
+    description="Display the tasks on the run queues of each cpu.",
+    arguments=(
+        mutually_exclusive_group(
+            argument("-t", action="store_true", dest="show_timestamps"),
+            argument("-T", action="store_true", dest="show_lag"),
+            argument("-m", action="store_true", dest="pretty_runtime"),
+            argument("-g", action="store_true", dest="group"),
+        ),
+        argument("-c", type=str, default="a", dest="cpus"),
+        drgn_argument,
+    ),
+)
+def _crash_cmd_runq(
+    prog: Program, name: str, args: argparse.Namespace, **kwargs: Any
+) -> None:
+    table_format = args.show_timestamps or args.show_lag or args.pretty_runtime
+    table: List[List[Any]] = []
+    headers: List[Any] = []
+
+    runq_clocks: Dict[int, int] = {}
+    cpus = parse_cpuspec(args.cpus).cpus(prog)
+    for i, (cpu, runqueue) in enumerate(get_rq_per_cpu(prog, cpus)):
+        curr_task = runqueue.curr[0].address_of_()
```

Comment:
What is the `[0].address_of_()` for? Instead, I think we want to read it:

```suggestion
        curr_task = runqueue.curr.read_()
```